### ISSUE TITLE
Difference between distribute and local machine version when training by tensorflow 

### ISSUE THREAD

**Issue by yananchen1989** [tensorflow/tensorflow#14494]
I use a common DNN network with less than 4 layers for a regression task. When I test my codes in local single machine with CPU, everything is fine. But when I train it on distribute system, which uses 30 machines' CPU and in the asynchronous parameters update way, all the layers' output values, including the last prediction values, come into Nan. The learning rate in local and distribute training are both 0.01. I changed it to 0.0001 in distribute way, it seems fine that the outputs are in normal range. I am confused about why learning rate cause this phenomenon. Any friend can try to explain it ? Thanks.

--
The label is a float, ranging from 0 to 1. Most of them falls into 0.1 to 0.3

**Comment by yaroslavvb** [comment]
To misquote Tolstoy -- "Happy training jobs are all alike, every unhappy job is unhappy in its own way." 

Async training jobs sometimes gets problems when gradients are too stale -- stale worker arriving with yesterday's gradient can throw whole procedure into NaN-land
